plugins/dialog/guest-js/index.ts

```diff
@@ -202,6 +301,16 @@ async function save(options: SaveDialogOptions = {}): Promise<string | null> {
   return await invoke('plugin:dialog|save', { options })
 }
 
+/**
+ * The result of a message dialog.
+ *
+ * The result is a string if the dialog has custom buttons,
+ * otherwise it is one of the default buttons.
+ *
+ * @since 2.3.0
+ */
+export type MessageDialogResult = 'Yes' | 'No' | 'Ok' | 'Cancel' | (string & {})
```

Comment:
Should these be in lower case?

Reply:
Not really, I think PascalCase is better here as they are sort of constants

Reply:
Asking because our `kind` is in lower case and things like `checkPermissions` returns this `type PermissionState = 'granted' | 'denied' | 'prompt' | 'prompt-with-rationale'`

Reply:
I still don't like them tbh, so I changed it to an Enum

Reply:
`MessageDialogResult` should stay the same tbh as it matches what `rfd` returns as the result for custom and sometimes built-in buttons.

Reply:
That makes sense

Reply:
> I still don't like them tbh, so I changed MessageDialogDefaultButtons to an Enum

To be honest that enum feels too lengthy than it could have been `await message('hi', { buttons: MessageDialogDefaultButtons.Ok })`, also typescript enums are kinda weird and have runtime costs that I would like to avoid most of the time

About the casing, I think this is what most of the web are using, and we should follow the convention

> https://w3ctag.github.io/design-principles/#casing-rules

Reply:
We shouldn't strictly follow these design principles as tauri is a mix of Rust and JS, I would prefer to have APIs consistent between JS and Rust, as it makes it easier to find and less cognitive load imo.

Reply:
> To be honest that enum feels too lengthy than it could have been `await message('hi', { buttons: MessageDialogDefaultButtons.Ok })`

I am fine to change it back to constant strings, but I would still keep them PascalCase

Reply:
Alright, let's just use the union type and keep the casing then, I don't mind it too much, just feel like that's more consistent with other APIs we currently have